### ISSUE TITLE
Attribute error

### ISSUE THREAD

**Issue by ixje** [CityOfZion/neo3-boa#230]
**Describe the bug**
Getting the following stack trace on the contract below
```
(venv) eriks-MBP:neo3-boa erik$ neo3-boa example2.py
INFO: Started compiling	example2.py
--- Logging error ---
Traceback (most recent call last):
  File "/Users/erik/Documents/code/neo3-boa/boa3/analyser/astanalyser.py", line 53, in visit
    return super().visit(node)
  File "/usr/local/Cellar/python@3.8/3.8.3/Frameworks/Python.framework/Versions/3.8/lib/python3.8/ast.py", line 363, in visit
    return visitor(node)
  File "/Users/erik/Documents/code/neo3-boa/boa3/analyser/astoptimizer.py", line 142, in visit_BinOp
    return self.parse_to_node(str(value), bin_op)
  File "/Users/erik/Documents/code/neo3-boa/boa3/analyser/astoptimizer.py", line 59, in parse_to_node
    new_node = self.visit(super().parse_to_node(expression, origin))
  File "/Users/erik/Documents/code/neo3-boa/boa3/analyser/astanalyser.py", line 137, in parse_to_node
    node = ast.parse(expression)
  File "/usr/local/Cellar/python@3.8/3.8.3/Frameworks/Python.framework/Versions/3.8/lib/python3.8/ast.py", line 47, in parse
    return compile(source, filename, mode, flags,
  File "<unknown>", line 1
    bye worldhi
        ^
SyntaxError: invalid syntax

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/Cellar/python@3.8/3.8.3/Frameworks/Python.framework/Versions/3.8/lib/python3.8/logging/__init__.py", line 1081, in emit
    msg = self.format(record)
  File "/usr/local/Cellar/python@3.8/3.8.3/Frameworks/Python.framework/Versions/3.8/lib/python3.8/logging/__init__.py", line 925, in format
    return fmt.format(record)
  File "/usr/local/Cellar/python@3.8/3.8.3/Frameworks/Python.framework/Versions/3.8/lib/python3.8/logging/__init__.py", line 664, in format
    record.message = record.getMessage()
  File "/usr/local/Cellar/python@3.8/3.8.3/Frameworks/Python.framework/Versions/3.8/lib/python3.8/logging/__init__.py", line 367, in getMessage
    msg = str(self.msg)
  File "/Users/erik/Documents/code/neo3-boa/boa3/exception/CompilerError.py", line 24, in __str__
    return self.message
  File "/Users/erik/Documents/code/neo3-boa/boa3/exception/CompilerError.py", line 16, in message
    message = '' if self._error_message is None else ' - ' + self._error_message
  File "/Users/erik/Documents/code/neo3-boa/boa3/exception/CompilerError.py", line 79, in _error_message
    message += ": %s" % self.raised_exception.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "/Users/erik/Documents/code/neo3-boa/venv/bin/neo3-boa", line 11, in <module>
    load_entry_point('neo3-boa', 'console_scripts', 'neo3-boa')()
  File "/Users/erik/Documents/code/neo3-boa/boa3/cli.py", line 23, in main
    Boa3.compile_and_save(args.input)
  File "/Users/erik/Documents/code/neo3-boa/boa3/boa3.py", line 43, in compile_and_save
    Compiler().compile_and_save(path, output_path, show_errors)
  File "/Users/erik/Documents/code/neo3-boa/boa3/compiler/compiler.py", line 46, in compile_and_save
    self.bytecode = self.compile(path, log)
  File "/Users/erik/Documents/code/neo3-boa/boa3/compiler/compiler.py", line 35, in compile
    self._analyse(fullpath, log)
  File "/Users/erik/Documents/code/neo3-boa/boa3/compiler/compiler.py", line 56, in _analyse
    self._analyser = Analyser.analyse(path, log)
  File "/Users/erik/Documents/code/neo3-boa/boa3/analyser/analyser.py", line 57, in analyse
    analyser.__pos_execute()
  File "/Users/erik/Documents/code/neo3-boa/boa3/analyser/analyser.py", line 98, in __pos_execute
    AstOptimizer(self, log=self._log)
  File "/Users/erik/Documents/code/neo3-boa/boa3/analyser/astoptimizer.py", line 36, in __init__
    self.visit(self._tree)
  File "/Users/erik/Documents/code/neo3-boa/boa3/analyser/astanalyser.py", line 53, in visit
    return super().visit(node)
  File "/usr/local/Cellar/python@3.8/3.8.3/Frameworks/Python.framework/Versions/3.8/lib/python3.8/ast.py", line 363, in visit
    return visitor(node)
  File "/usr/local/Cellar/python@3.8/3.8.3/Frameworks/Python.framework/Versions/3.8/lib/python3.8/ast.py", line 439, in generic_visit
    value = self.visit(value)
  File "/Users/erik/Documents/code/neo3-boa/boa3/analyser/astanalyser.py", line 53, in visit
    return super().visit(node)
  File "/usr/local/Cellar/python@3.8/3.8.3/Frameworks/Python.framework/Versions/3.8/lib/python3.8/ast.py", line 363, in visit
    return visitor(node)
  File "/Users/erik/Documents/code/neo3-boa/boa3/analyser/astoptimizer.py", line 78, in visit_FunctionDef
    super().generic_visit(node)
  File "/usr/local/Cellar/python@3.8/3.8.3/Frameworks/Python.framework/Versions/3.8/lib/python3.8/ast.py", line 439, in generic_visit
    value = self.visit(value)
  File "/Users/erik/Documents/code/neo3-boa/boa3/analyser/astanalyser.py", line 53, in visit
    return super().visit(node)
  File "/usr/local/Cellar/python@3.8/3.8.3/Frameworks/Python.framework/Versions/3.8/lib/python3.8/ast.py", line 363, in visit
    return visitor(node)
  File "/usr/local/Cellar/python@3.8/3.8.3/Frameworks/Python.framework/Versions/3.8/lib/python3.8/ast.py", line 448, in generic_visit
    new_node = self.visit(old_value)
  File "/Users/erik/Documents/code/neo3-boa/boa3/analyser/astanalyser.py", line 60, in visit
    self._log_error(
  File "/Users/erik/Documents/code/neo3-boa/boa3/analyser/astanalyser.py", line 43, in _log_error
    logging.error(error)
Message: InternalError()
Arguments: ()
ERROR: 'Return' object has no attribute 'value'
Traceback (most recent call last):
  File "/Users/erik/Documents/code/neo3-boa/boa3/cli.py", line 23, in main
    Boa3.compile_and_save(args.input)
  File "/Users/erik/Documents/code/neo3-boa/boa3/boa3.py", line 43, in compile_and_save
    Compiler().compile_and_save(path, output_path, show_errors)
  File "/Users/erik/Documents/code/neo3-boa/boa3/compiler/compiler.py", line 46, in compile_and_save
    self.bytecode = self.compile(path, log)
  File "/Users/erik/Documents/code/neo3-boa/boa3/compiler/compiler.py", line 36, in compile
    return self._compile()
  File "/Users/erik/Documents/code/neo3-boa/boa3/compiler/compiler.py", line 67, in _compile
    return CodeGenerator.generate_code(self._analyser)
  File "/Users/erik/Documents/code/neo3-boa/boa3/compiler/codegenerator.py", line 49, in generate_code
    visitor.visit(analyser.ast_tree)
  File "/usr/local/Cellar/python@3.8/3.8.3/Frameworks/Python.framework/Versions/3.8/lib/python3.8/ast.py", line 363, in visit
    return visitor(node)
  File "/Users/erik/Documents/code/neo3-boa/boa3/compiler/codegeneratorvisitor.py", line 100, in visit_Module
    self.visit(stmt)
  File "/usr/local/Cellar/python@3.8/3.8.3/Frameworks/Python.framework/Versions/3.8/lib/python3.8/ast.py", line 363, in visit
    return visitor(node)
  File "/Users/erik/Documents/code/neo3-boa/boa3/compiler/codegeneratorvisitor.py", line 163, in visit_FunctionDef
    self.visit_to_map(stmt)
  File "/Users/erik/Documents/code/neo3-boa/boa3/compiler/codegeneratorvisitor.py", line 52, in visit_to_map
    value = self.visit(node)
  File "/usr/local/Cellar/python@3.8/3.8.3/Frameworks/Python.framework/Versions/3.8/lib/python3.8/ast.py", line 363, in visit
    return visitor(node)
  File "/Users/erik/Documents/code/neo3-boa/boa3/compiler/codegeneratorvisitor.py", line 200, in visit_Return
    if ret.value is not None:
AttributeError: 'Return' object has no attribute 'value'
```

**How to Reproduce**
try to compile this contract
```python
def main() -> str:
    return "bye world" + "hi"
```

**Expected behavior**
either working contract or clean compiler error

**Screenshots**

**Platform:**
 - OS: OSX 
 - Version: latest master branch commit 3b653597e5c0504ff310d37fe5e4e19be3f01b02
 - Python version: 3.8

**Comment by meevee98** [comment]
It seems that #235 has fixed the root cause of this. This attribute error was caused by the optimization, that couldn't differentiate string values from code statements. In this case, it was evaluating ``"bye worldhi"`` as a Python code.

To prevent other unexpected errors to throw the stack trace to the user, I've changed the construction of `InternalError`

**Comment by ixje** [comment]
confirmed to be solved on dev branch (e795f82048de7f4b7f2c9228c3abe18730f38579)
